src/test/java/io/supertokens/test/passwordless/PasswordlessConsumeCodeTest.java

```diff
@@ -0,0 +1,801 @@
+            return;
+        }
+
+        PasswordlessStorage storage = StorageLayer.getPasswordlessStorage(process.getProcess());
+
+        Passwordless.CreateCodeResponse createCodeResponse = Passwordless.createCode(process.getProcess(), EMAIL, null,
+                null, null);
+        assertNotNull(createCodeResponse);
+
+        Exception error = null;
+        int maxCodeInputAttempts = Config.getConfig(process.getProcess()).getPasswordlessMaxCodeInputAttempts();
+
+        for (int counter = 0; counter < maxCodeInputAttempts; counter++) {
+
+            try {
+                Passwordless.consumeCode(process.getProcess(), createCodeResponse.deviceId,
+                        createCodeResponse.deviceIdHash, null, "n0p321");
+            } catch (Exception ex) {
+                error = ex;
+            }
+
+        }
+
+        assertNotNull(error);
+        assert (error instanceof RestartFlowException);
+
+        // verify that devices have not been cleared
+        PasswordlessDevice[] devices = storage.getDevicesByEmail(EMAIL);
+        assertNotEquals(0, devices.length);
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    /**
+     * link with too many failedAttempts (changed maxCodeInputAttempts configuration between consumes)
+     * TODO: review -> do we need to create code again post restart ?
+     *
+     * @throws Exception
+     */
+
+    @Test
+    public void testConsumeWrongUserInputCodeExceedingMaxAttemptsWithConfigUpdate() throws Exception {
+        // start process with default configuration for max attempt
+        TestingProcessManager.TestingProcess process = startApplicationWithDefaultArgs();
+        PasswordlessStorage storage = StorageLayer.getPasswordlessStorage(process.getProcess());
+
+        Passwordless.CreateCodeResponse createCodeResponse = Passwordless.createCode(process.getProcess(), EMAIL, null,
+                null, null);
+        assertNotNull(createCodeResponse);
+
+        Exception error = null;
+        int maxCodeInputAttempts = Config.getConfig(process.getProcess()).getPasswordlessMaxCodeInputAttempts();
+
+        // input wrong code max attempts - 2 times
+        for (int counter = 0; counter < maxCodeInputAttempts - 2; counter++) {
+
+            try {
+                Passwordless.consumeCode(process.getProcess(), createCodeResponse.deviceId,
+                        createCodeResponse.deviceIdHash, "n0p321", null);
+            } catch (Exception ex) {
+
+            }
+
+        }
+
+        // verify that devices have not been cleared
+        PasswordlessDevice[] devices = storage.getDevicesByEmail(EMAIL);
+        assertNotEquals(0, devices.length);
+
+        // kill process and restart with increased max code input attempts
+        killApplication(process);
+
+        String[] args = { "../" };
+        Utils.setValueInConfig("passwordless_max_code_input_attempts", "7");
```

Comment:
updated it to be relative to the original value. One question though, do we need to create code again post restart ? Because it throws 'RestartFlowException' irrespective of the value of passwordless_max_code_input_attempts.

Reply:
Irrespective of `passwordless_max_code_input_attempts`? That might be a bug, I think we should discuss this on a call tomorrow